### DISCUSSION TITLE
Should the path of the `image` function escape special characters?

### DISCUSSION THREAD

**Discussion by lettuceDestroyer** [typst/typst#3992] [Q&A] [answered]
I recently tried to add a figure in a typst file like this:
```typst
#figure(image("assets\test\some_image.png"))
```
which threw the error: ` file not found (searched at ...\assetsest\some_image.png) `

The reason my path contains backward slashes (`\`) instead of a forward slash (`/`) is because I am getting the path with the VSCode right click menu `Copy relative path` This results in me either having to escape what I believe to be the escape character, change the settings in VSCode to use a forward slash as separator or change them manually.

The same thing happens with `\n` and `\r` (maybe more as I didn't check all possible combinations).

I'm just gonna assume that `\t` is parsed to a tab, `r` to carriage return and `\n` to new line.

My question is what use case is there where that behaviour would be wanted in a path?

**Comment by MDLC01** [comment, marked as answer]
The problem is that the argument to `image` is just a regular string, and there are good reasons to allow escape sequences in string literals. There is a proposal for a separate `path` type (#971). If this came with its own syntax, it would likely not allow escape sequences.

**Comment by Enivex** [comment]
Note that you can do

```
`assets\test\some_image.png`.text
```
instead of using escape sequences.
